File: 1786-count-the-number-of-consistent-strings/count-the-number-of-consistent-strings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        result = 0

        for word in words:
            characters = set(list(word))
            for i in characters:
                if i not in allowed:
                    break
            else:
                result += 1
        
        else:
            return result

File: 1786-count-the-number-of-consistent-strings/count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        vector<int> is_allowed(26);
        for (auto c : allowed) {
            is_allowed[c - 'a'] = 1;
        }
        int result = 0;
        for (auto x : words) {
            bool flag = true;
            for (auto c : x) {
                if (!is_allowed[c - 'a']) {
                    flag = false;
                    break;
                }
            }
            if (flag)
                result++;
        }
        return result;
    }
};